mlflow/pytorch/pytorch_autolog.py

```diff
@@ -1,180 +1,156 @@
+
+        def on_test_end(self, trainer, pl_module):
+            """
+            Logs accuracy and other relevant metrics on the testing end
+            """
+            mlflow.set_tag(key="Mode", value="testing")
+            for key, value in trainer.callback_metrics.items():
+                try_mlflow_log(mlflow.log_metric, key, float(value))
+
+        @staticmethod
+        def _log_early_stop_params(early_stop_obj):
+            """
+            Logs Early Stop parameters into mlflow
+            """
+            if hasattr(early_stop_obj, "monitor"):
+                try_mlflow_log(mlflow.log_param, "monitor", early_stop_obj.monitor)
+
+            if hasattr(early_stop_obj, "mode"):
+                try_mlflow_log(mlflow.log_param, "mode", early_stop_obj.mode)
+
+            if hasattr(early_stop_obj, "patience"):
+                try_mlflow_log(mlflow.log_param, "patience", early_stop_obj.patience)
+
+            if hasattr(early_stop_obj, "min_delta"):
+                try_mlflow_log(mlflow.log_param, "min_delta", early_stop_obj.min_delta)
+
+            if hasattr(early_stop_obj, "stopped_epoch"):
+                try_mlflow_log(mlflow.log_param, "stopped_epoch", early_stop_obj.stopped_epoch)
+
+        @staticmethod
+        def _early_stop_check(early_stop_callback):
+            """
+            Logs all early stopping metrics
+            """
+            if early_stop_callback.stopped_epoch != 0:
+
+                if hasattr(early_stop_callback, "stopped_epoch"):
+                    try_mlflow_log(mlflow.log_metric, "Stopped_Epoch", early_stop_callback.stopped_epoch)
+                    restored_epoch = early_stop_callback.stopped_epoch - max(
+                        1, early_stop_callback.patience
+                    )
+                    try_mlflow_log(mlflow.log_metric, "Restored_Epoch", restored_epoch)
 
-        tempdir = tempfile.mkdtemp()
-        if os.path.exists(tempdir):
-            shutil.rmtree(tempdir)
-
-        try:
-            mlflow.pytorch.save_model(trainer.model, path=tempdir)
-            trainer.logger.experiment.log_artifact(
-                trainer.logger.run_id, local_path=tempdir, artifact_path="model"
-            )
-        finally:
-            shutil.rmtree(tempdir)
-
-        if trainer.early_stop_callback and trainer.checkpoint_callback.best_model_path:
-            trainer.logger.experiment.log_artifact(
-                trainer.logger.run_id,
-                local_path=trainer.checkpoint_callback.best_model_path,
-                artifact_path="restored_model_checkpoint",
-            )
-
-    def on_test_end(self, trainer, pl_module):
-        """
-        Logs accuracy and other relevant metrics on the testing end
-        """
-        metrics = trainer.callback_metrics
+                if hasattr(early_stop_callback, "best_score"):
```

Comment:
Please add details of all the early_stop_callback, early_stop_obj and all other optional parameters that user can specify for autologing in the Readme